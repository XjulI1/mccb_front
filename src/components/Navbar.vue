<template>
  <div class="btn-group main-navbar" role="group">
    <all-account-button/>
    <virement-button/>
    <new-operation-button/>
    <retrait-button/>
    <params-button/>
  </div>
</template>

<script>
  import NewOperationButton from './NavBar/NewOperationButton'
  import VirementButton from './NavBar/VirementButton'
  import RetraitButton from './NavBar/RetraitButton'
  import ParamsButton from './NavBar/ParamsButton'
  import AllAccountButton from './NavBar/AllAccountButton'

  export default {
    name: 'navbar',
    components: { AllAccountButton, ParamsButton, RetraitButton, VirementButton, NewOperationButton },
    data () {
      return {}
    },

    created () {
    },

    methods: {}
  }
</script>

<style lang="scss" scoped>
  @import '@/global.scss';

  .main-navbar {
    padding: 0;

    @media screen and (max-width: 768px) {
      height: $navbar-height;
      width: 100%;
      background-color: lightgrey;
      position: fixed;
      bottom: 0;
      left: 0;
      display: inline;
      text-align: center;
      z-index: 10;
    }
  }
</style>
