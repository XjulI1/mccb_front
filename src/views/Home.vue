<template>
  <div class="home">
    <account-header/>
    <operation-list/>
    <navbar/>
  </div>
</template>

<script>
  import Navbar from '../components/Navbar'
  import AccountHeader from '../components/AccountHeader'
  import { mapState } from 'vuex'
  import OperationList from '../components/OperationList'

  export default {
    name: 'home',
    components: { OperationList, AccountHeader, Navbar },

    beforeMount () {
      this.$store.dispatch('fetchUser', 1)
        .then(() => {
          return this.$store.dispatch('fetchActiveAccount', this.$store.state.user.favoris)
        })
        .then(() => {
          this.$store.dispatch('fetchOperationsOfActiveAccount')
        })
    },

    data () {
      return {}
    },

    computed: mapState(['user', 'activeAccount']),

    created () {
    },

    methods: {}
  }
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style lang="scss" scoped>
  @import '@/global.scss';

  .home {
    margin-top: $header-account-height + 5;
    margin-bottom: $navbar-height;
  }
</style>
