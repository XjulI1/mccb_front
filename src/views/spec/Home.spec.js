import { createLocalVue, shallowMount } from '@vue/test-utils'

import Router from 'vue-router'

import Home from '@/views/Home'
import Navbar from '@/components/Navbar'

const localVue = createLocalVue()

localVue.use(Router)

describe('Home.vue', () => {
  test('should render correct contents', () => {
    const wrapper = shallowMount(Home, {
      localVue
    })

    expect(wrapper.findAll(Navbar).length).toBe(1)
  })

  test('should match snapshot', () => {
    const wrapper = shallowMount(Home, {
      localVue
    })

    expect(wrapper.element).toBeDefined()
    expect(wrapper.element).toMatchSnapshot()
  })
})
