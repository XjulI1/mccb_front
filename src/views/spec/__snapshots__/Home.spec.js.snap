// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.vue should match snapshot 1`] = `
<div
  class="home"
>
  <router-view-stub
    name="default"
  />
   
  <accountheader-stub />
   
  <div
    class="left-panel"
  >
    <comptelist-stub />
     
    <navbar-stub />
  </div>
   
  <operationlist-stub />
</div>
`;
